### PR TITLE
[Feature]: Dark Mode

### LINKED ISSUE #87: [FEATURE] Add a dark mode to the website
### Detailed description

I think a dark mode should be added to the website. I currently use Dark Reader to make the websites dark, but for Dev Protocol, it doesn't look that good, IMO.

image

### Context

Many users do not like the light mode, they prefer dark mode (like me 😂), so I think it'd be a good idea to add a dark mode to the website.

### Possible implementation

Add a dark mode to the website, that can be toggled with a button in the navbar.

### Additional information

Nope.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by AbhinavMV**
#### Description of the change

Added dark mode

Closes #87 
#### Checklist

**I agree to the following :-**

*   Added description of the change
*   I've read the [contributing guidelines](https://github.com/web3community/devprotocol.xyz/blob/main/CONTRIBUTING.md)
*   Search previous suggestions before making a new PR, as yours may be a duplicate.
*   I acknowledge that all my contributions will be made under the project's license.

Notes (**optional**): 
Need help to fix following issues:
1. Unable to change dropdown svg color to white in dark mode (text-white class not working don't know why as it is an svg)
2. Toggle theme button animation works on small screen but not on large screen. (I made different file for toggle button and i am using this component 2 times in navbar.vue file maybe because of this it is not working)

Additionally 
In Dapps section images are used instead of svg so in dark mode they don't look good. (I don't think we need to change this as i can see an issue is raised to change this section itself)

<a href="https://gitpod.io/#https://github.com/web3community/devprotocol.xyz/pull/145"></a>

### CODE REVIEW

File: src/components/Global/ThemeButton.vue [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  data() {
+    return {
+      isDarkMode: false,
+    };
+  },
+  methods: {
+    toggleDarkMode() {
+      // var checkbox = document.getElementById('#toggle')
+      if (this.isDarkMode) {
+        localStorage.removeItem("theme");
```
Comment (krishguptadev) [inline src/components/Global/ThemeButton.vue:58]: How about we also implement a media query for prefers-color-scheme

**Review by krishguptadev — CHANGES_REQUESTED** [review]

File: tailwind.config.js [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-      }
-    }
-  }
-}
+        para: "40ch",
+        half: "50%",
+      },
+    },
+  },
+};
```
Comment (KukretiShubham) [inline tailwind.config.js:21]: Codacy has a fix for the issue: [Extra semicolon.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
}
```

File: tailwind.config.js [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+        body: ['"Whyte"', "sans-serif"],
       },
       maxWidth: {
-        para: '40ch',
-        half: '50%'
-      }
-    }
-  }
-}
+        para: "40ch",
```
Comment (KukretiShubham) [inline tailwind.config.js:16]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        para: '40ch',
```

File: tailwind.config.js [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
+      colors: {
+        darkSecondary: "#151414",
+      },
       fontFamily: {
-        mono: ['IBM Plex Mono', 'monospace'],
-        display: ['WhyteInktrap', 'sans-serif'],
-        body: ['"Whyte"', 'sans-serif']
+        mono: ["IBM Plex Mono", "monospace"],
+        display: ["WhyteInktrap", "sans-serif"],
+        body: ['"Whyte"', "sans-serif"],
```
Comment (KukretiShubham) [inline tailwind.config.js:13]: Codacy has a fix for the issue: [Unexpected trailing comma.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        body: ['"Whyte"', "sans-serif"]
```

File: tailwind.config.js [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
       },
       maxWidth: {
-        para: '40ch',
-        half: '50%'
-      }
-    }
-  }
-}
+        para: "40ch",
+        half: "50%",
```
Comment (KukretiShubham) [inline tailwind.config.js:17]: Codacy has a fix for the issue: [Unexpected trailing comma.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        half: "50%"
```

Comment (KukretiShubham) [inline tailwind.config.js:17]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        half: '50%',
```

File: tailwind.config.js (hunk 4 above)
Comment (KukretiShubham) [inline tailwind.config.js:13]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        body: ['"Whyte"', 'sans-serif'],
```

File: tailwind.config.js [hunk 6]
```diff
@@ -1,17 +1,21 @@
 module.exports = {
-  mode: 'jit',
-  purge: ['./public/**/*.html', './src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}'],
+  mode: "jit",
+  purge: ["./public/**/*.html", "./src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}"],
+  darkMode: "class",
   theme: {
     extend: {
+      colors: {
+        darkSecondary: "#151414",
```
Comment (KukretiShubham) [inline tailwind.config.js:8]: Codacy has a fix for the issue: [Unexpected trailing comma.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        darkSecondary: "#151414"
```

File: tailwind.config.js [hunk 7]
```diff
@@ -1,17 +1,21 @@
...
+      },
       fontFamily: {
-        mono: ['IBM Plex Mono', 'monospace'],
-        display: ['WhyteInktrap', 'sans-serif'],
-        body: ['"Whyte"', 'sans-serif']
+        mono: ["IBM Plex Mono", "monospace"],
+        display: ["WhyteInktrap", "sans-serif"],
```
Comment (KukretiShubham) [inline tailwind.config.js:12]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        display: ['WhyteInktrap', "sans-serif"],
```

File: tailwind.config.js [hunk 8]
```diff
@@ -1,17 +1,21 @@
 module.exports = {
-  mode: 'jit',
-  purge: ['./public/**/*.html', './src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}'],
+  mode: "jit",
+  purge: ["./public/**/*.html", "./src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}"],
```
Comment (KukretiShubham) [inline tailwind.config.js:3]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
  purge: ['./public/**/*.html', "./src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}"],
```

File: tailwind.config.js [hunk 9]
```diff
@@ -1,17 +1,21 @@
...
   theme: {
     extend: {
+      colors: {
+        darkSecondary: "#151414",
+      },
       fontFamily: {
-        mono: ['IBM Plex Mono', 'monospace'],
-        display: ['WhyteInktrap', 'sans-serif'],
-        body: ['"Whyte"', 'sans-serif']
+        mono: ["IBM Plex Mono", "monospace"],
```
Comment (KukretiShubham) [inline tailwind.config.js:11]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        mono: ['IBM Plex Mono', "monospace"],
```

File: tailwind.config.js [hunk 10]
```diff
@@ -1,17 +1,21 @@
...
       maxWidth: {
-        para: '40ch',
-        half: '50%'
-      }
-    }
-  }
-}
+        para: "40ch",
+        half: "50%",
+      },
```
Comment (KukretiShubham) [inline tailwind.config.js:18]: Codacy has a fix for the issue: [Unexpected trailing comma.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
      }
```

File: tailwind.config.js (hunk 6 above)
Comment (KukretiShubham) [inline tailwind.config.js:8]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        darkSecondary: '#151414',
```

File: tailwind.config.js [hunk 11]
```diff
@@ -1,17 +1,21 @@
 module.exports = {
-  mode: 'jit',
-  purge: ['./public/**/*.html', './src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}'],
+  mode: "jit",
+  purge: ["./public/**/*.html", "./src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}"],
+  darkMode: "class",
```
Comment (KukretiShubham) [inline tailwind.config.js:4]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
  darkMode: 'class',
```

File: tailwind.config.js [hunk 12]
```diff
@@ -1,17 +1,21 @@
 module.exports = {
-  mode: 'jit',
-  purge: ['./public/**/*.html', './src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}'],
+  mode: "jit",
```
Comment (KukretiShubham) [inline tailwind.config.js:2]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
  mode: 'jit',
```

File: src/components/Global/ThemeButton.vue (hunk 1 above)
Comment (AbhinavMV) [reply src/components/Global/ThemeButton.vue:58]: prefers-color-scheme is used in mounted().
When page is loaded at the beginning

**Review by krishguptadev — CHANGES_REQUESTED** [review]
Also, check the comments left by @KukretiShubham (codacy suggestions)

File: tailwind.config.js [hunk 13]
```diff
@@ -1,17 +1,21 @@
 module.exports = {
   mode: 'jit',
-  purge: ['./public/**/*.html', './src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}'],
+  purge: ['./public/**/*.html', "./src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}"],
+  darkMode: 'class',
   theme: {
     extend: {
+      colors: {
+        darkSecondary: "#151414"
```
Comment (KukretiShubham) [inline tailwind.config.js:8]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        darkSecondary: '#151414'
```

File: tailwind.config.js [hunk 14]
```diff
@@ -1,17 +1,21 @@
 module.exports = {
   mode: 'jit',
-  purge: ['./public/**/*.html', './src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}'],
+  purge: ['./public/**/*.html', "./src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}"],
```
Comment (KukretiShubham) [inline tailwind.config.js:3]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
  purge: ['./public/**/*.html', './src/**/*.{astro,js,jsx,svelte,ts,tsx,vue}'],
```

File: tailwind.config.js [hunk 15]
```diff
@@ -1,17 +1,21 @@
...
     extend: {
+      colors: {
+        darkSecondary: "#151414"
+      },
       fontFamily: {
-        mono: ['IBM Plex Mono', 'monospace'],
-        display: ['WhyteInktrap', 'sans-serif'],
-        body: ['"Whyte"', 'sans-serif']
+        mono: ['IBM Plex Mono', "monospace"],
+        display: ['WhyteInktrap', "sans-serif"],
```
Comment (KukretiShubham) [inline tailwind.config.js:12]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        display: ['WhyteInktrap', 'sans-serif'],
```

File: tailwind.config.js [hunk 16]
```diff
@@ -1,17 +1,21 @@
...
+        body: ['"Whyte"', "sans-serif"]
```
Comment (KukretiShubham) [inline tailwind.config.js:13]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        body: ['"Whyte"', 'sans-serif']
```

File: tailwind.config.js [hunk 17]
```diff
@@ -1,17 +1,21 @@
...
       },
       maxWidth: {
         para: '40ch',
-        half: '50%'
+        half: "50%"
       }
-    }
-  }
+    },
```
Comment (KukretiShubham) [inline tailwind.config.js:19]: Codacy has a fix for the issue: [Unexpected trailing comma.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
      }
```

File: tailwind.config.js [hunk 18]
```diff
@@ -1,17 +1,21 @@
...
   theme: {
     extend: {
+      colors: {
+        darkSecondary: "#151414"
+      },
       fontFamily: {
-        mono: ['IBM Plex Mono', 'monospace'],
-        display: ['WhyteInktrap', 'sans-serif'],
-        body: ['"Whyte"', 'sans-serif']
+        mono: ['IBM Plex Mono', "monospace"],
```
Comment (KukretiShubham) [inline tailwind.config.js:11]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
        mono: ['IBM Plex Mono', 'monospace'],
```

File: tailwind.config.js [hunk 19]
```diff
@@ -12,6 +16,6 @@ module.exports = {
         para: '40ch',
         half: '50%'
       }
-    }
-  }
+      }
```
Comment (KukretiShubham) [inline tailwind.config.js:19]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 6.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
    }
```

File: tailwind.config.js [hunk 20]
```diff
@@ -12,6 +16,6 @@ module.exports = {
...
+  },
```
Comment (KukretiShubham) [inline tailwind.config.js:20]: Codacy has a fix for the issue: [Unexpected trailing comma.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
      }
```

**Review by krishguptadev — CHANGES_REQUESTED** [review]
The toggle for light disappears in dark mode.
image

File: tailwind.config.js [hunk 21]
```diff
@@ -13,5 +17,5 @@ module.exports = {
         half: '50%'
       }
     }
-  }
+      }
```
Comment (KukretiShubham) [inline tailwind.config.js:20]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 6.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
    }
```

**Review by krishguptadev — CHANGES_REQUESTED** [review]
The logos aren't accessible 
image

**Review by krishguptadev — CHANGES_REQUESTED** [review]
Same here...

image

**Review by krishguptadev — CHANGES_REQUESTED** [review]
The background of body is too dark, can it be like 222222 (HEX)

File: tailwind.config.js [hunk 22]
```diff
@@ -13,5 +17,5 @@ module.exports = {
         half: '50%'
       }
     }
-  }
+    }
```
Comment (KukretiShubham) [inline tailwind.config.js:20]: Codacy has a fix for the issue: [Expected indentation of 4 spaces but found 6.](https://app.codacy.com/gh/web3community/devprotocol.xyz/pullRequest?prid=8426368)
```suggestion
    }
```

**Review by krishguptadev — CHANGES_REQUESTED** [review]
There is no animation if I click more than once. 
image

**Review by krishguptadev — CHANGES_REQUESTED** [review]
Can you run eslint once?

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please see the comments above.

**Comment by AbhinavMV** [comment]
> The logos aren't accessible image

What would you suggest to fix this? 
making background white of each image makes it look bad.

**Comment by AbhinavMV** [comment]
> The background of body is too dark, can it be like 222222 (HEX)

sure will make this change 👍

**Comment by AbhinavMV** [comment]
> The toggle for light disappears in dark mode. image

Yes i know i mentioned this in the PR.
Need help to fix following issues:
Unable to change dropdown svg color to white in dark mode (text-white class not working don't know why as it is an svg)
Toggle theme button animation works on small screen but not on large screen. (I made different file for toggle button and i am using this component 2 times in navbar.vue file maybe because of this it is not working)
Additionally
In Dapps section images are used instead of svg so in dark mode they don't look good. (I don't think we need to change this as i can see an issue is raised to change this section itself)

**Comment by vinzvinci** [comment]
I think Dapps section should be change first? check #131

**Review by KukretiShubham — COMMENTED** [review]
Toggle doesn't work, though it switches modes.

**Comment by AbhinavMV** [comment]
> I think Dapps section should be change first? check #131

ok Will take a look at it.

**Comment by AbhinavMV** [comment]
> Toggle doesn't work, though it switches modes.

Fixed it 👍. Completely changed the button design.

**Review by vinzvinci — COMMENTED** [review]
I love the toggle button ❤️

**Comment by shuklaritvik06** [comment]
Yep toggle button looks great now ✨

**Review by Panquesito7 — APPROVED** [review]
Awesome! 🚀 LGTM! I love this. 😄👍

**Review by krishguptadev — APPROVED** [review]

**Review by krishguptadev — COMMENTED** [review]
## Merge Conflict

Thank you for your contribution. Your pull request currently has conflicting changes with our `main` branch. In order to review this pull request, we will need those conflicts to be resolved.

We recommend doing this with the `git rebase` command. If you are unsure of how to fix this, let us know and we would be happy to assist you.

**Review by KukretiShubham — APPROVED** [review]

**Review by vinzvinci — COMMENTED** [review]
Thanks, everyone! LGTM

**Pull request merged**